packages/core/android/src/main/java/io/embrace/embracewrapper/EmbraceManagerModule.java

```diff
@@ -562,6 +568,110 @@ public void logNetworkClientError(String url,
+        try{
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().startSpan(name, parentSpanId));
+        }catch(Exception e){
+            promise.resolve(null);
+        }
+    }
+
+    private Map<String, String> convertToReadableMap(ReadableMap readableMap) {
+        if (readableMap == null) {
+            return null;
+        }
+
+        Map<String, String> stringMap = new HashMap<>();
+
+        for (String key : readableMap.toHashMap().keySet()) {
+            stringMap.put(key, readableMap.getString(key));
+        }
+
+        return stringMap;
+    }
+
+    private ErrorCode getSpanErrorCodebyString(String errorCode){
+        switch (errorCode) {
+            case "Failure":
+                return ErrorCode.FAILURE;
+
+            case "UserAbandon":
+                return ErrorCode.USER_ABANDON;
+
+            case "Unknown":
+                return ErrorCode.UNKNOWN;
+
+            default:
+                return null;
+        }
+    }
+
+    private List<Map<String, Object>> transformListReadableMapToListMap(ReadableArray items) {
+        List<Map<String, Object>> objectMapList = new ArrayList<>();
+
+        try{
+            for (int i = 0; i < items.size(); i++) {
+                ReadableMap readableMap = items.getMap(i);
+
+                if (readableMap != null) {
+                    Map<String, Object> map = readableMap.toHashMap();
+                    // TODO Change when Android/iOS replace time in nano for ms
+                    if (map.containsKey("timestampNanos") && map.get("timestampNanos") instanceof Double) {
+                        double timestampNanos = (Double) map.get("timestampNanos");
+                        map.put("timestampNanos", (long) timestampNanos);
+                    }
+                    objectMapList.add(map);
+                }
+            }
+        }catch(Exception e){
+            Log.e("Embrace", "There was an error in parsing the span event data", e);
+        }
+
+        return objectMapList;
+    }
+
+    @ReactMethod()
+    public void stopSpanWithId(String spanId, String errorCodeString, Promise promise) {
+        try{
+            
+            ErrorCode errorCodeInstance = this.getSpanErrorCodebyString(errorCodeString);
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().stopSpan(spanId, errorCodeInstance));
+        }catch(Exception e){
+            promise.resolve(false);
+        }
+    }
+
+    @ReactMethod()
+    public void addSpanAttributesToSpanId(String spanId, String key, String value, Promise promise) {
+        try{
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().addSpanAttribute(spanId, key, value));
+        }catch(Exception e){
+            promise.resolve(false);
+        }
+    }
+
+    @ReactMethod()
+    public void addSpanEventToSpanId(String spanId, String name, Double time, ReadableMap attributes, Promise promise) {
```

Comment:
Same comment as above with the method name, re: ToSpan vs WithSpanId

Reply:
I implemented iOS first and they call it that way. should I change ? 
<img width="438" alt="image" src="https://github.com/embrace-io/embrace-react-native-sdk/assets/33431136/7c498af9-1177-42c4-9e50-5d3f35adc207">